Add (regression) E2E test to make sure the line height setting is present

fullofcaffeine opened the pull request:
## Changes proposed in this Pull Request

Closes #46880

Adds a regression test in the client block E2E spec to make sure the `line-height` setting is preset for paragraphs and avoid the regression of this issue: https://github.com/Automattic/wp-calypso/issues/46880.

### How to test

1. Setup your env to run Calypso E2E tests (search for "Automated end-to-end Testing") in the FG;
1. Edit `./node_modules/.bin/mocha specs/wp-calypso-gutenberg-post-editor-spec.js` and call `.only` after `describe` for this test in order to restrict the spec to it;
1. Run with `./node_modules/.bin/mocha specs/wp-calypso-gutenberg-post-editor-spec.js`;
1. It should pass 🟢 

#### Negative scenario

1. Setup your env to run Calypso E2E tests (search for "Automated end-to-end Testing") in the FG;
1. Update your WPCOM sandbox with the latest changes from upstream, make sure you're running at least GB 9.2;
1. Open `wp-calypso/test/e2e/config/local-decrypted.json`, find the entry for `gutenbergSimpleSiteUser`, grab the WP test site address from there
1. Map this address to your sandbox
1. ssh into your sandbox
1. Open `full-site-site-editing.php` in `mu-plugins` and find the `get_current_prod_version` function. Grab the value of the `$__autogen_plugin_version_prod` var;
1. Now go to the `ET` directory in the `plugins` directory and open the right version directory. Edit `common/index.php`. Remove or comment-out line 155 (`add_theme_support( 'custom-line-height' );`). Exit your sandbox;
1. Edit `./node_modules/.bin/mocha specs/wp-calypso-gutenberg-post-editor-spec.js` and call `.only` after `describe` for this test in order to restrict the spec to it;
1. Run the spec with `./node_modules/.bin/mocha specs/wp-calypso-gutenberg-post-editor-spec.js`;
1. It should fail 🔴 


### Observations and questions

1. I'm still not sure if this test belongs in that spec (`wp-calypso-gutenberg-post-editor-spec.js`) or if it should be moved somewhere else. Still pondering. Any hints?;
1. Should we test this for the page editor as well? (I assume not, since the behavior should be the same);
1. I'm just checking for the presence of the div that wraps the setting UI control, which is enough for the purposes of the regression test.

### TODO 

- [x] Deal with the mobile viewport, tests are failing

fullofcaffeine committed:
Add (regression) e2e test to make sure the line height setting is show in the settings pane for a paragraph block

fullofcaffeine committed:
Fix test on the mobile viewport

Opening the settings pane on mobile seems to make the editor DOM element inaccessible. Inputting the text into the editor *before* opening the settings pane fixes the issue.

WunderBart commented on test/e2e/specs/wp-calypso-gutenberg-post-editor-spec.js:
```diff
@@ -1296,4 +1297,26 @@ describe( `[${ host }] Calypso Gutenberg Editor: Posts (${ screenSize })`, funct
+				driver,
+				By.css( '.block-editor-line-height-control' )
+			);
+			assert.strictEqual( lineHeighSettingPresent, true, 'Line height setting not found' );
```
I think this line is equal to:

```js
assert.ok( lineHeightSettingPresent, 'Line height setting not found.' );
```

Just a tiny nitpick - feel free to ignore 👍

fullofcaffeine replied:
Ah, that's much leaner, thanks for the suggestion!

WunderBart commented on test/e2e/specs/wp-calypso-gutenberg-post-editor-spec.js:
```diff
@@ -1296,4 +1297,26 @@ describe( `[${ host }] Calypso Gutenberg Editor: Posts (${ screenSize })`, funct
+			await loginFlow.loginAndStartNewPost( null, true );
+		} );
+
+		step( 'Line height setting is available', async function () {
```
I think that instead of creating a separate `Settings` section which feels pretty vague, it would be better to have this under the [`Edit a post`](https://github.com/Automattic/wp-calypso/blob/57e257d9482c7b0c91b2f1033f2700dcbafa1587/test/e2e/specs/wp-calypso-gutenberg-post-editor-spec.js#L861) one. Sth like:

```js
describe( 'Edit a Post: @parallel', function () {
    describe( 'Paragraph settings', function () {
        step( 'Line height is available', function () {
                // ...
        } );
    } );
} );
```

What do you think?

fullofcaffeine replied:
That makes sense! I'll change 👍🏻

WunderBart reviewed (approved):
> I'm still not sure if this test belongs in that spec (wp-calypso-gutenberg-post-editor-spec.js) or if it should be moved somewhere else. Still pondering. Any hints?

I think it fits well in that general suite, but I would move the under the [`Edit a post`](https://github.com/Automattic/wp-calypso/blob/57e257d9482c7b0c91b2f1033f2700dcbafa1587/test/e2e/specs/wp-calypso-gutenberg-post-editor-spec.js#L861) section, as mentioned [here](https://github.com/Automattic/wp-calypso/pull/47429#discussion_r530471111).

> Should we test this for the page editor as well? (I assume not, since the behavior should be the same)

Yeah, I think it should be the same - no need to add another test!

> I'm just checking for the presence of the div that wraps the setting UI control, which is enough for the purposes of the regression test.

Agreed, thanks!

---

Overall it works well for me, thanks for adding that test! I've left just 2 comments. No blockers though!

fullofcaffeine committed:
Move line-height test inside the "Edit a Post" test as a step, simplify assertion

Take advantage of the environment created by previous steps, including not having to login again, saving some precious time.

fullofcaffeine commented:
@WunderBart Thanks a lot for taking the time to review and the suggestions!

I figured we could simplify it further by moving this check as a step thus taking advantage of the context already created by previous steps (including not having to login again!) resulting in less code needed and saving some precious time in process.

fullofcaffeine committed:
fixup! Remove line break

fullofcaffeine committed:
Fix mobile test due to lingering settings popup getting on the way of the editor

On mobile, the settings popup will prevent access to the editor. The way the test is setup means state from previous steps will affect the next one and so on, so we need to make sure to close the popup before manipulating anything in the block editor.

fullofcaffeine commented:
`test-e2e-canary-woo` seems to be flaky at the moment. It's not related to the changes in this changeset.
